### ISSUE TITLE
Display Loading Indicator During Language Translation Process

### ISSUE THREAD

**Issue by nguaman** [QuiiBz/next-international#405]
**Is your feature request related to a problem? Please describe.**

My file with the translations is a fairly large JSON, so the first time I switch languages it takes a few seconds to load. Therefore, I would like to display a message indicating that the text is being translated. I tried using useEffect based on the locale (useCurrentLocale), but it only detects when it starts fetching the translation and not when it finishes, so I have no way to remove the message indicating that the translation is in progress.

**Describe the solution you'd like**
Expose some state of the translation process.

`const [isTranslationInProgress, changeLocale] = useChangeLocale();`

**Describe alternatives you've considered**
Having some way to detect when the translation is finished or an alternative solution.

**Additional context**
I tried using Suspense to detect when it loads but couldn't find a way to make it work.

**Comment by gustaveWPM** [comment]
> I tried using useEffect based on the locale (useCurrentLocale), but it only detects when it starts fetching the translation and not when it finishes

Yes, the locale will be updated only when the file is loaded to avoid flickering. You can't rely on it. :/
